src/main/java/org/opensearch/flowframework/model/WorkflowState.java

```diff
@@ -0,0 +1,466 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.model;
+
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.flowframework.util.ParseUtils;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.flowframework.util.ParseUtils.parseStringToStringMap;
+
+/**
+ * The WorkflowState is used to store all additional information regarding a workflow that isn't part of the
+ * global context.
+ */
+public class WorkflowState implements ToXContentObject {
+    /** The template field name for the associated workflowID **/
+    public static final String WORKFLOW_ID_FIELD = "workflow_id";
+    /** The template field name for the workflow error **/
+    public static final String ERROR_FIELD = "error";
+    /** The template field name for the workflow state **/
+    public static final String STATE_FIELD = "state";
+    /** The template field name for the workflow provisioning progress **/
+    public static final String PROVISIONING_PROGRESS_FIELD = "provisioning_progress";
+    /** The template field name for the workflow provisioning start time **/
+    public static final String PROVISION_START_TIME_FIELD = "provision_start_time";
+    /** The template field name for the workflow provisioning end time **/
+    public static final String PROVISION_END_TIME_FIELD = "provision_end_time";
+    /** The template field name for the user who created the workflow **/
+    public static final String USER_FIELD = "user";
+    /** The template field name for the workflow ui metadata **/
+    public static final String UI_METADATA_FIELD = "ui_metadata";
+    /** The template field name for template user outputs */
+    public static final String USER_OUTPUTS_FIELD = "user_outputs";
+    /** The template field name for template resources created */
+    public static final String RESOURCES_CREATED_FIELD = "resources_created";
```

Comment:
Can we move all these fields to CommonValue.java?

Reply:
No problem doing this, it just seems like for all the classes in the `model` package we left the field names as part of that class, could crowd the CommonValue. No strong opinion here but should I move all field strings across all models we have and will have.

Reply:
Yeah let's move fields to CommonValue.java.